torch/_refs/nn/functional/__init__.py

```diff
@@ -332,6 +333,27 @@ def tanhshrink(a: TensorLikeType) -> TensorLikeType:
     return refs.sub(a, refs.tanh(a))
 
 
+@register_decomposition(torch.ops.aten.threshold)
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a",),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+)
+def threshold(
+    a: TensorLikeType, threshold: NumberType, value: NumberType, inplace: bool = False
+) -> TensorLikeType:
+    """
+    Reference implementation of torch.nn.functional.threshold
+    """
+
+    if inplace:
+        raise NotImplementedError
+
+    # the straightforward computation drops nan's from input in some cases
+    partial = torch.where(a > threshold, a, value)
```

Comment:
@mruberry i think threshold does not support complex numbers (based on trial with existing eager), but there doesn't seem to be an analogue of 'NumberType' that doesn't include complex.  Should I just assert value is not complex and hope that is enough to make mypy happy, or is there any other pattern i can follow here?

Reply:
You can do that, and/or you an explicitly annotate the type with `Union[int, float]` or `Union[bool, int, float]` as appropriate, too